assets/js/base/context/cart-checkout/checkout/processor/index.js

```diff
@@ -1,12 +1,111 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import triggerFetch from '@wordpress/api-fetch';
+import {
+	useCheckoutContext,
+	useShippingDataContext,
+	useBillingDataContext,
+	usePaymentMethodDataContext,
+} from '@woocommerce/base-context';
+import { useEffect, useRef, useCallback } from '@wordpress/element';
+import { useStoreNotices } from '@woocommerce/base-hooks';
+
 /**
  * CheckoutProcessor component. @todo Needs to consume all contexts.
  *
- * Subscribes to checkout context and triggers processing when needed:
- * 		1. Create an order (draft) when email address is provided
- * 		2. Process an order when the place order button is pressed
- * 		   (checkout processing)
+ * Subscribes to checkout context and triggers processing via the API.
  */
 const CheckoutProcessor = () => {
+	const { onCheckoutProcessing, dispatchActions } = useCheckoutContext();
+	const { shippingAddress } = useShippingDataContext();
+	const { billingData } = useBillingDataContext();
+	const { activePaymentMethod } = usePaymentMethodDataContext();
+	const { addErrorNotice } = useStoreNotices();
+	const currentBillingData = useRef( billingData );
+	const currentShippingAddress = useRef( shippingAddress );
+
+	useEffect( () => {
+		currentBillingData.current = billingData;
+		currentShippingAddress.current = shippingAddress;
+	}, [ billingData, shippingAddress ] );
+
+	/**
+	 * Process an order via the /wc/store/checkout endpoint.
+	 *
+	 * @return {boolean} True if everything was successful.
+	 */
+	const processCheckout = useCallback( async () => {
+		await triggerFetch( {
+			path: '/wc/store/checkout',
+			method: 'POST',
+			data: {
+				payment_method: activePaymentMethod,
+				payment_data: [],
```

Comment:
Is including the payment data  in the checkout processing to be done in a follow-up?

Reply:
Depends; is it ready to go now via the provider?

Reply:
Well I'm mostly concerned that we could _miss_ this. As long as it's documented for a follow-up I'm fine with it happening after this pull is merged.